tests/LazyBelongsToManyTest.php

```diff
@@ -2,6 +2,25 @@
 
 namespace Tests;
 
-class LazyBelongsToManyTest
+use Tests\Models\Post;
+use Tests\Models\Tag;
+
+class LazyBelongsToManyTest extends TestCase
 {
+    public function testLazyLoading()
+    {
+        $post = Post::find(1);
+
+        $this->assertEquals(2, $post->tags->count());
```

Comment:
It would be better if we use the query on the relationship and not on the collection.

Reply:
I tried to use the relation query instead but it gets unexpected values. 
Anyway, I will inspect it and try to solve this problem and get back to ASAP !

Reply:
So it means I need to fix it. I'll Merge it and then try to work on that at ease. Thanks!!